### PR TITLE
change: init-governance command

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

* adds `run_init_governance` function that submits the transaction initializing on-chain governance
* adds a `partner-chains-smart-contracts-commands` crate used by the node to add smart contract-related commands
* adds first command `init-governance` 

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages.
- [ ] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [ ] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: toolkit/utils/ogmios-client/src/types.rs [hunk 1]
```diff
@@ -33,6 +33,12 @@ pub struct OgmiosUtxo {
 	pub script: Option<OgmiosScript>,
 }
 
+impl OgmiosUtxo {
+	pub fn id_string(&self) -> String {
```
Comment (LGLO) [inline toolkit/utils/ogmios-client/src/types.rs:37]: Cannot it be Display trait instead?

File: node/node/Cargo.toml [hunk 2]
```diff
@@ -69,6 +69,7 @@ thiserror = { workspace = true }
 time-source = { workspace = true }
 derive-new = { workspace = true }
 partner-chains-node-commands = { workspace = true }
+partner-chains-smart-contracts-commands = { workspace = true }
```
Comment (LGLO) [inline node/node/Cargo.toml:72]: To be honest, I've expected that `partner-chains-node-commands` will bring it and users will need one our crate for commands, not two

File: toolkit/cli/smart-contracts-commands/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#[derive(Clone, Debug, clap::Parser)]
+pub enum SmartContractsCmdStandalone {
+	#[clap(flatten)]
+	Inner(SmartContractsCmd),
+}
+
+type CmdResult<T> = Result<T, Box<dyn std::error::Error + Send + Sync>>;
+
+#[tokio::main]
+async fn main() -> CmdResult<()> {
```
Comment (LGLO) [inline toolkit/cli/smart-contracts-commands/src/main.rs:13]: Do we need it? The plan was to have zero binaries. Users were supposed use their own node binary

File: toolkit/offchain/Cargo.toml [hunk 4]
```diff
@@ -15,22 +15,24 @@ anyhow = { workspace = true }
...
 pallas-primitives = { workspace = true }
-partner-chains-plutus-data = { workspace = true }
 plutus = { workspace = true }
 raw-scripts = { workspace = true }
 serde = { workspace = true }
 serde_json = { workspace = true, features = ["std"] }
 secp256k1 = { workspace = true, features = ["std"] }
 sidechain-domain = { workspace = true, features = ["std"] }
 uplc = { workspace = true }
+tokio = { workspace = true }
```
Comment (LGLO) [inline toolkit/offchain/Cargo.toml:35]: Could you remove it from dev-dependencies, please?

File: toolkit/offchain/src/init_governance/mod.rs [hunk 5]
```diff
@@ -1,4 +1,85 @@
...
+		cost,
+	)?;
+	let signed_transaction = tx_context.sign(unsigned_transaction);
+
+	let result = client.submit_transaction(&signed_transaction.to_bytes()).await?;
+
+	println!("✅ Transaction submited. ID: {}", hex::encode(result.transaction.id));
+
+	Ok(())
+}
```
Comment (LGLO) [inline toolkit/offchain/src/init_governance/mod.rs:85]: Apparently there is no test for this method

File: node/node/Cargo.toml (hunk 2 above)
Comment (AmbientTea) [reply node/node/Cargo.toml:72]: it does. I removed this dependency

File: toolkit/cli/smart-contracts-commands/src/main.rs (hunk 3 above)
Comment (AmbientTea) [reply toolkit/cli/smart-contracts-commands/src/main.rs:13]: That's QoL for us, because the node crate takes a long time to rebuild during development.

Comment (LGLO) [reply toolkit/cli/smart-contracts-commands/src/main.rs:13]: Okay, I won't block PR on this. I'll try to figure out some alternative, when I work on other commands

**Pull request merged**